cvat/apps/auto_segmentation/views.py

```diff
@@ -86,33 +112,37 @@ class InferenceConfig(coco.CocoConfig):
 
     ## RUN OBJECT DETECTION
     result = {}
-    frames = frame_provider.get_frames(frame_provider.Quality.ORIGINAL)
-    for image_num, (image_bytes, _) in enumerate(frames):
+    image_num = 0
+    image_list = list(frame_provider.get_frames(frame_provider.Quality.ORIGINAL))
```

Comment:
Could there be memory issue here? Why do you need cast to `list` here?

Reply:
We didn't face any issues while testing, but yes I agree with you. We need a length of the frame_provider to figure out distribution among available GPUs. The Generator class does not have length function. Let me see if we can have an alternative way. Do you have any suggestions?

Reply:
We didn't face any issue while testing, but I agree with you. We need a length of a frame_provider to distribute images among all available gpus. The generator class does not have length, so used it. Let me see if we can come up with an alternative. Do you have any suggestions?

Reply:
`frame_provider` has `__len__` definition https://github.com/opencv/cvat/blob/develop/cvat/apps/engine/frame_provider.py#L83

Reply:
thanks, I will update this.